Translate docs/node-interface (#62)
gatsbyjs/gatsby-pl: docs/docs/node-interface.md

title: Node Interface
---

> "Node" jest centrum systemu danych Gatsby. Wszystkie dane które są dodane do Gatsby są modelowane wykorzystując node'y.

Reviewer: Just small typo. Please use comma ',' before 'które'.

  fieldOwners: Object,
  content: String,
}
...innne pola określone dla tego typu node'a
```

### `parent`

> Klucz zarezerwowany dla wtyczek które chcą rozszerzyć swoje node'y.

Reviewer: Just small typo. Please use comma ',' before 'które'.

### `contentDigest`

Skrót "Hash", lub krótkie cyfrowe streszczenie zawartości tego node'a (np. `md5sum`).

Skrót powinien być unikatowy dla zawartości tego node'a, ponieważ jest on używany do buforowania. Jeśli zawartość zmieni się, skrót również powinien ulec zmianie. Istnieje funkcja pomocnicza o nazwie [createContentDigest](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-core-utils/src/create-content-digest.js), która służy do utworzenia skrótu `md5`.

### `mediaType`

Opcjonalny [typ nośnika](https://pl.wikipedia.org/wiki/Typ_MIME) do wskazania wtyczek przekształcających ten node oraz jego dane, które mogą być dalej przetwarzane.

### `type`

Globalny, unikatowy typ node'a wybrany przez właściciela wtyczki.

### `owner`

Wtyczka, przez którą stworzony został node. To pole jest dodawane przez Gatsby (a nie przez wtyczkę).

### `fieldOwners`

Magazynuje informacje, które wtyczki utworzyły poszczególne pola. To pole jest dodawane przez Gatsby (a nie przez wtyczkę).

### `content`

> Opcjonalne pole dostarczające surową zawartość dla tego node'a którą mogą pobierać wtyczki przekształcające w celu dalszego przetwarzania.

Reviewer: Just small typo. Please use comma ',' before 'którą'.

## Wtyczki źródłowe

> Nowe node'y dodawane są do Gatsby poprzez wtyczki "źródłowe". Powszechnie stosowaną przez wiele witryn Gatsby wtyczką źródłową jest [wtyczka źródłowa systemu plików](/packages/gatsby-source-filesystem/) która przekształca pliki na dysku na node'y pliku.

Reviewer: Just small typo. Please use comma ',' before 'która'.

Inne wtyczki źródłowe zaciągają dane z zewnętrznych interfejsów API takich jak np. [Drupal](/packages/gatsby-source-drupal/) czy [Hacker News](/packages/gatsby-source-hacker-news/).

## Wtyczki przekształcające

Wtyczki przekształcające mogą również tworzyć node'y poprzez przekształcenie źródeł node'ów w nowe typy node'ów. Instalowanie obu wtyczek - źródłowych i przekształcających jest powszechną praktyką podczas budowania stron Gatsby.

Node'y utworzone przez wtyczki przekształcające są ustawione jako "dzieci" swoich "rodziców".

- [Wtyczka przekształcająca dane (biblioteka Markdown)](/packages/gatsby-transformer-remark/) wyszukuje nowych node'ów,
  które zostały utworzone z typem `text/markdown` dla `mediaType` i  
  przekształca je na node'y `MarkdownRemark` z polami `html`.
- [Wtyczka przekształcająca YAML](/packages/gatsby-transformer-yaml/) wyszukuje nowe node'y z typem nośnika `text/yaml` (np. pliki `.yaml`) i tworzy nowe YAML dzieci node'y poprzez transformację źródła YAML w obiekt JavaScript.

## GraphQL

> Gatsby automatycznie wprowadza strukturę node'ów Twojej strony i tworzy schemat GraphQL, który można następnie odpytywać z komponentów twojej strony.

Reviewer: Uniform style, "Twojej" vs "twojej". Preferred 'twojej' IMO, because of docs.

  fieldOwners: Object,
  content: String,
}
...innne pola określone dla tego typu node'a
```

### `parent`

> Klucz zazrezerwowany dla wtyczek które chcą rozszerzyć swoje node'y.

Reviewer 2: ```suggestion
Klucz zarezerwowany dla wtyczek które chcą rozszerzyć swoje node'y.
```

### `contentDigest`

Skrót "Hash", lub krótkie cyfrowe streszczenie zawartości tego node'a (np. `md5sum`).

Skrót powinien być unikatowy dla zawartości tego node'a, ponieważ jest on używany do buforowania. Jeśli zawartość zmieni się, skrót również powinien ulec zmianie. Istnieje funkcja pomocnicza o nazwie [createContentDigest](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-core-utils/src/create-content-digest.js), która służy do utworzenia skrótu `md5`.

### `mediaType`

Opcjonalny [typ nośnika](https://pl.wikipedia.org/wiki/Typ_MIME) do wskazania wtyczek przekształcających ten node oraz jego dane, które mogą być dalej przetwarzane.

### `type`

Globalny, unikatowy typ node'a wybrany przez właściciela wtyczki.

### `owner`

> Wtyczka przez którą stworzony został node. To pole jest dodawane przez Gatsby(a nie przez wtyczkę).

Reviewer 2: ```suggestion
Wtyczka, przez którą stworzony został node. To pole jest dodawane przez Gatsby (a nie przez wtyczkę).
```

### `fieldOwners`

> Magazyn informacji które wtyczki utworzyły poszczególne pola. To pole jest dodawane przez Gatsby(a nie przez wtyczkę).

Reviewer 2: ```suggestion
Magazynuje informacje, które wtyczki utworzyły poszczególne pola. To pole jest dodawane przez Gatsby (a nie przez wtyczkę).
```

### `content`

Opcjonalne pole dostarczające surową zawartość dla tego node'a którą mogą pobierać wtyczki przekształcające w celu dalszego przetwarzania.

## Wtyczki źródłowe

Nowe node'y dodawane są do Gatsby poprzez wtyczki "źródłowe". Powszechnie stosowaną przez wiele witryn Gatsby wtyczką źródłową jest [wtyczka źródłowa systemu plików](/packages/gatsby-source-filesystem/) która przekształca pliki na dysku na node'y pliku.

Inne wtyczki źródłowe zaciągają dane z zewnętrznych interfejsów API takich jak np. [Drupal](/packages/gatsby-source-drupal/) czy [Hacker News](/packages/gatsby-source-hacker-news/).

## Wtyczki przekształcające

Wtyczki przekształcające mogą również tworzyć node'y poprzez przekształcenie źródeł node'ów w nowe typy node'ów. Instalowanie obu wtyczek - źródłowych i przekształcających jest powszechną praktyką podczas budowania stron Gatsby.

Node'y utworzone przez wtyczki przekształcające są ustawione jako "dzieci" swoich "rodziców".

> - [Wtyczka przekształcająca dane(biblioteka Markdown)](/packages/gatsby-transformer-remark/) wyszukuje nowych node'ów,

Reviewer 2: ```suggestion
- [Wtyczka przekształcająca dane (biblioteka Markdown)](/packages/gatsby-transformer-remark/) wyszukuje nowych node'ów,
```

  które zostały utworzone z typem `text/markdown` dla `mediaType` i  
  przekształca je na node'y `MarkdownRemark` z polami `html`.

> - [Wtyczka przekształcająca YAML](/packages/gatsby-transformer-yaml/) wyszukuje nowych node'ów z typem
>   nośnika `text/yaml` (np. pliki `.yaml`) i tworzy nowe YAML dzieci node'a/node'ów poprzez
>   transformację źródła YAML w obiekt JavaScript.

Reviewer 2: ```suggestion
- [Wtyczka przekształcająca YAML](/packages/gatsby-transformer-yaml/) wyszukuje nowe node'y z typem nośnika `text/yaml` (np. pliki `.yaml`) i tworzy nowe YAML dzieci node'y poprzez transformację źródła YAML w obiekt JavaScript.
```

### `parent`

> Klucz zazrezerwowany dla wtyczek, które chcą rozszerzyć swoje node'y.

Reviewer: Typo: zazrezerwowany

Please fix to 'zarezerwowany'
